Classes/ExpressionLanguage/DocumentTypeFunctionProvider.php

```diff
@@ -132,25 +134,26 @@ function($arguments, $cPid)
                     return $type;
                 }
 
+                // 3d object type if model parameter is not empty
+                if (!empty($queryParams['tx_dlf']['model'])) {
+                    return TYPE_OBJECT;
+                }
+
                 // Load document with current plugin parameters.
                 $this->loadDocument($queryParams['tx_dlf'], $cPid);
                 if (!isset($this->document) || $this->document->getCurrentDocument() === null) {
                     return $type;
                 }
+
                 // Set PID for metadata definitions.
                 $this->document->getCurrentDocument()->cPid = $cPid;
 
                 $metadata = $this->document->getCurrentDocument()->getToplevelMetadata($cPid);
-                if (!empty($metadata['type'][0])) {
-                    // Calendar plugin does not support IIIF (yet). Abort for all newspaper related types.
-                    if (
-                        $this->document->getCurrentDocument() instanceof IiifManifest
-                        && array_search($metadata['type'][0], ['newspaper', 'ephemera', 'year', 'issue']) !== false
-                    ) {
-                        return $type;
-                    }
+                if (!empty($metadata['type'][0])
+                    && (!$this->isIiifManifestWithNewspaperRelatedType($metadata['type'][0]) || $metadata['type'][0] == TYPE_OBJECT)) {
```

Comment:
```suggestion
                    && !$this->isIiifManifestWithNewspaperRelatedType($metadata['type'][0])) {
```
We don't need the additional clause. `$this->isIiifManifestWithNewspaperRelatedType()` already returns `false` for 3D objects (which is negated and therefore `true`).